WordPress/Classes/Services/SharingService.swift

```diff
@@ -0,0 +1,410 @@
+        let remote = SharingServiceRemote(api: apiForRequest())
+
+        remote.getPublicizeServices( {(remoteServices:[RemotePublicizeService]) -> Void in
+            // Process the results
+            self.mergePublicizeServices(remoteServices, success: success)
+        },
+        failure: { (error: NSError!) -> Void in
+            failure?(error)
+        })
+    }
+
+
+    /**
+    Fetches the current user's list of keyring connections. Nothing is saved to core data.
+    The success block should accept an array of `KeyringConnection` objects.
+
+    @param success An optional success block accepting an array of `KeyringConnection` objects
+    @param failure An optional failure block accepting an `NSError` parameter
+    */
+    public func fetchKeyringConnections(success: ([KeyringConnection] -> Void)?, failure: (NSError! -> Void)?) {
+        let remote = SharingServiceRemote(api: apiForRequest())
+
+        remote.getKeyringConnections( {(keyringConnections:[KeyringConnection]) -> Void in
+            // Just return the result
+            success?(keyringConnections)
+        },
+        failure: { (error: NSError!) -> Void in
+            failure?(error)
+        })
+    }
+
+
+    /**
+    Syncs Publicize connections for the specified wpcom blog.
+
+    @param blog The `Blog` for which to sync publicize connections
+    @param success An optional success block accepting no parameters.
+    @param failure An optional failure block accepting an `NSError` parameter.
+    */
+    public func syncPublicizeConnectionsForBlog(blog:Blog, success: (() -> Void)?, failure: (NSError! -> Void)?) {
+        let blogObjectID = blog.objectID
+        let remote = SharingServiceRemote(api: apiForRequest())
+        remote.getPublicizeConnections(blog.dotComID, success: {(remoteConnections:[RemotePublicizeConnection]) -> Void in
+            // Process the results
+            self.mergePublicizeConnectionsForBlog(blogObjectID, remoteConnections:remoteConnections, success: success)
+        },
+        failure: { (error: NSError!) -> Void in
+            failure?(error)
+        })
+    }
+
+
+    /**
+    Creates a new publicize connection for the specified `Blog`, using the specified
+    keyring.  Optionally the connection can target a particular external user account.
+
+    @param blog The `Blog` for which to sync publicize connections
+    @param keyring The `KeyringConnection` to use
+    @param externalUserID An optional string representing a user ID on the external service.
+    @param success An optional success block accepting a `PublicizeConnection` parameter.
+    @param failure An optional failure block accepting an NSError parameter.
+    */
+    public func createPublicizeConnectionForBlog(blog:Blog,
+        keyring:KeyringConnection,
+        externalUserID:String?,
+        success: (PublicizeConnection -> Void)?,
+        failure: (NSError! -> Void)?)
+    {
+
+            let blogObjectID = blog.objectID
```

Comment:
I think it looks that way because of the placement of the `{`. Stacked parameters makes for tricky alignment.